plugins/dbviewer/frontend/public/javascripts/countly.views.js

```diff
@@ -218,12 +218,19 @@ window.DBViewerView = countlyView.extend({
             $('#aggregate-view').show();
             $('#aggregate-header').show();
             $('#dbviewer-header').hide();
+            $('.dbviewer-prepare-area').show();
+            $('#aggregate-result-table_wrapper').hide();
             $('#generate_aggregate_report').text($.i18n.map["dbviewer.generate-aggregate-report"]);
-            $('#back_to_dbviewer').css('display', 'block');
+            $('#back_to_dbviewer').css('display', 'inline-block');
         });
 
         $('body').off('click', '#back_to_dbviewer').on('click', '#back_to_dbviewer', function() {
-            app.navigate('#/manage/db/' + self.db + '/' + self.collection, true);
+            if (self._task) {
+                app.navigate('#/manage/db/' + store.get('dbviewer_current_db') + '/' + store.get('dbviewer_current_collection'), true);
```

Comment:
You should really stop using storage for state purposes and find another way to pass parameters, which is not persistent.
And do we really need them here? if you store the data in self.fb and self.collection from tasks meta data, they would already be set there and no need changing this method, wouldn't it?